go/utils.go

```diff
@@ -157,3 +157,25 @@ func WaitForTask(taskRef xenapi.TaskRef, delay int) error {
 	}
 	return nil
 }
+
+func WaitForSRReady(srRef xenapi.SRRef) error {
+	token := ""
+	maxTries := 3
+	eventTypes := []string{"PBD"}
+	for i := 0; i < maxTries; i++ {
+		eventBatch, err := xenapi.Event.From(session, eventTypes, token, 10.0)
+		if err != nil {
+			return err
+		}
+		token = eventBatch.Token
+		for _, event := range eventBatch.Events {
+			if event.Class == "pbd"  {
+				pbdRecord := event.Snapshot.(map[string]interface{})
+			    if pbdRecord["SR"].(string) == string(srRef) && pbdRecord["currently_attached"].(bool) == true {
```

Comment:
The indent seems not right. Can this go through the go ci?

Reply:
I think this function can be generalized since the accessing is on a map instead of specific structs:
`WaitUntil(session, event_types, pred_fun)` 
But this could be improved when another similar use case comes in.